### ISSUE TITLE
HTCondor devs request a mock plugin for integration testing

### ISSUE THREAD

**Issue by jhiemstrawisc** [PelicanPlatform/pelican#919]
The HTCondor developers who are working on integrating the pelican plugin have asked that we work on a mock plugin they can use for testing. This would allow them to write their own set of integration tests on top of the plugin.

**Comment by bbockelm** [comment]
What does this mean though?  A plugin that simply responds with the expected output to stdout?  That seems minimally useful (and definitely would go into the HTCondor codebase, not Pelican).

What would be more useful is being able to launch a mini-federation inside their development container and using a real, not a mock, plugin.

**Comment by johnkn** [comment]
We would want the plugin to respond with a "correct" output ad that either has success or a specific failure indicated by the input somehow.  This would be used in testing HTCondor to make sure the the way condor handles the success and failure of the plugin is correct.
'

**Comment by bbockelm** [comment]
So just a sample output from a success and failure?  E.g., a bash script that responds with one of a small collection of example output ClassAds?

**Comment by joereuss12** [comment]
Our "federation in a box" should serve as this "mock plugin" after conversations with condor devs. Closing this issue as test is currently underway.

**Comment by johnkn** [comment]
I don't believe that the request was a mock for integration testing.  I think the request was a mock for testing condor that could run in or batlab tests which won't have the option to install federation in a box before running the tests.

**Comment by joereuss12** [comment]
I thought when I talked to Tim T he said Pelican should be installed where these tests will be run so we should be able to issue Pelican commands for tests? I could be wrong in my understanding tho or he didn't completely understand what I was asking.

**Comment by bbockelm** [comment]
Closing as not planned due to inactivity.
